pkg/util/ciutil/az_pipelines.go

```diff
@@ -35,22 +35,12 @@ func (az azurePipelinesCI) DetectVars() Vars {
 	v.SHA = os.Getenv("BUILD_SOURCEVERSION")
 	v.BranchName = os.Getenv("BUILD_SOURCEBRANCHNAME")
 	v.CommitMessage = os.Getenv("BUILD_SOURCEVERSIONMESSAGE")
-	// Azure Pipelines can be connected to external repos.
-	// So we check if the provider is GitHub, then we use
-	// `SYSTEM_PULLREQUEST_PULLREQUESTNUMBER` instead of `SYSTEM_PULLREQUEST_PULLREQUESTID`.
-	// The PR ID/number only applies to Git repos.
-	vcsProvider := os.Getenv("BUILD_REPOSITORY_PROVIDER")
-	switch vcsProvider {
-	case "TfsGit":
-		orgURI := os.Getenv("SYSTEM_PULLREQUEST_TEAMFOUNDATIONCOLLECTIONURI")
-		projectName := os.Getenv("SYSTEM_TEAMPROJECT")
-		v.BuildURL = fmt.Sprintf("%v/%v/_build/results?buildId=%v", orgURI, projectName, v.BuildID)
-		// TfsGit is a git repo hosted on Azure DevOps.
-		v.PRNumber = os.Getenv("SYSTEM_PULLREQUEST_PULLREQUESTID")
-	case "GitHub":
-		// GitHub is a git repo hosted on GitHub.
-		v.PRNumber = os.Getenv("SYSTEM_PULLREQUEST_PULLREQUESTNUMBER")
-	}
+
+	orgURI := os.Getenv("SYSTEM_TEAMFOUNDATIONCOLLECTIONURI")
+	projectName := os.Getenv("SYSTEM_TEAMPROJECT")
+	v.BuildURL = fmt.Sprintf("%v/%v/_build/results?buildId=%v", orgURI, projectName, v.BuildID)
+
+	v.PRNumber = os.Getenv("SYSTEM_PULLREQUEST_PULLREQUESTNUMBER")
```

Comment:
For `GitHub`-backed VCS repos we need to use `SYSTEM_PULLREQUEST_PULLREQUESTNUMBER`, for other external Git repos, as well as Az DevOps' own Git repos, both `SYSTEM_PULLREQUEST_PULLREQUESTNUMBER` and `SYSTEM_PULLREQUEST_PULLREQUESTID` are the same. So we can just check this env var always. It is empty anyway if the repo is not a Git-based repo or if the build is not a PR build.